src/factories/mail-transport-factory.ts

```diff
@@ -0,0 +1,26 @@
+import { Transporter } from "nodemailer";
+import nodemailer from 'nodemailer';
+
+interface MailTransportFactoryData {
+  host?: string;
+  port?: number;
+  secure?: boolean;
+  auth?: {
+    user?: string;
+    pass?: string;
+  }
+}
+
+export function mailTransportFactory(data: MailTransportFactoryData): Transporter {
```

Comment:
Acredito que trocar o nome da interface para `TransporterOptions` além de deixar mais intuitivo fica mais curto.

```suggestion
interface TransporterOptions {
  host?: string;
  port?: number;
  secure?: boolean;
  auth?: {
    user?: string;
    pass?: string;
  }
}

export function mailTransportFactory(data: TransporterOptions): Transporter {
```